test/async-test-args.js

```diff
@@ -0,0 +1,167 @@
+const fs = require('fs');
+const path = require('path');
+const assert = require('chai').assert;
+const protagonist = require('./protagonist');
+
+const data = fs.readFileSync(path.join(__dirname, './fixtures/sample-api.apib'), 'utf8');
+const sync_parsed = protagonist.parseSync(data)
+const sync_parsed_sm = protagonist.parseSync(data, { exportSourcemap : true })
+
+describe('Async parse correctly handling options', () => {
+
+  it('will export source maps', function() {
+    protagonist.parse(data, { exportSourcemap : true })
+      .then(res => {
+        assert.deepEqual(res, sync_parsed_sm);
+      })
+      .catch(err => {
+        assert.fail()
+      })
+  })
+
+  it('will not export source maps', function() {
+    protagonist.parse(data, { exportSourcemap : false })
+      .then(res => {
+        assert.deepEqual(res, sync_parsed);
+      })
+      .catch(err => {
+        assert.fail()
+      })
+  })
+
+  it('will not accept empty options', function() {
+    protagonist.parse(data, {})
+      .then(res => {
+        assert.deepEqual(res, sync_parsed);
+      })
+      .catch(err => {
+        assert.fail()
+      })
+  })
+
+})
+
+
+describe('Async parsing - accepted parameters', () => {
+
+  const source = '# API\n ## GET /\n+ Response 200 (text/plain)\n        Hello world\n'
+  var opt = {}
+
+  describe('Promise', () => {
+    it('will accept source only', (done) => {
+        protagonist.parse(data)
+        .then(res => {
+           assert.isNotNull(res)
+           done()
+        })
+        .catch(err => {
+          done(err)
+        })
+    })
+
+    it('will accept source with options', (done) => {
```

Comment:
This test has already been done above, we can remove this `it`.

Reply:
Agree, it is already tested in section above.
But this set of tests is target to cover all possible combination of arguments
So I mean it has place there

Reply:
Let's not repeat the tests but try to make them more isolated. If we do this, it's hard to understand what's actually failing if something is wrong.